samples/Common.c

```diff
@@ -470,6 +470,69 @@ STATUS handleRemoteCandidate(PSampleStreamingSession pSampleStreamingSession, PS
     return retStatus;
 }
 
+STATUS lookForSslCert(PSampleConfiguration* ppSampleConfiguration)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    struct stat pathStat;
+    UINT32 length = 0;
+    DIR *dp;
+    struct dirent *entry;
+    BOOL certFound = FALSE;
+    PSampleConfiguration pSampleConfiguration = *ppSampleConfiguration;
+
+    pSampleConfiguration->pCaCertPath = getenv(CACERT_PATH_ENV_VAR);
+
+    // if ca cert path is not set from the environment, try to use the one that cmake detected
+    if (pSampleConfiguration->pCaCertPath == NULL) {
+        CHK_ERR(STRNLEN(DEFAULT_KVS_CACERT_PATH, MAX_PATH_LEN) > 0, STATUS_INVALID_OPERATION, "No ca cert path given (error:%s)", strerror(errno));
+        pSampleConfiguration->pCaCertPath = DEFAULT_KVS_CACERT_PATH;
+    }
+
+    else
+    {
+        // Check if the environment variable is a path
+        CHK(0 == FSTAT(pSampleConfiguration->pCaCertPath, &pathStat), STATUS_DIRECTORY_ENTRY_STAT_ERROR);
+
+        // If CaCertPath is a directory, check for a file with .pem extension in this directory
+        if(S_ISDIR(pathStat.st_mode)) {
+            CHK_ERR((dp = opendir(pSampleConfiguration->pCaCertPath)) != NULL, STATUS_DIRECTORY_OPEN_FAILED, "Cannot open directory");
+
+            while((entry = readdir(dp)) != NULL) {
+                if(entry->d_type != DT_DIR) {
+                    // Get length of detected filename
+                    length = (UINT32) STRLEN(entry->d_name);
+
+                    if((0 == STRCMPI(CA_CERT_PEM_FILE_EXTENSION, &entry->d_name[length - ARRAY_SIZE(CA_CERT_PEM_FILE_EXTENSION) + 1]))) {
```

Comment:
You can dereference past entry->d_name array. Ensure you check the length before deref

Reply:
Good point. Missed that. Doing it now.

Reply:
Added length check.